pkg/storage/unified/sql/db/dbimpl/db_engine_test.go

```diff
@@ -0,0 +1,301 @@
+	var prefix string
+	if withKeyPrefix {
+		prefix = "db_"
+	}
+	return newTestConfGetter(map[string]string{
+		prefix + "type":     dbTypeMySQL,
+		prefix + "host":     "/var/run/mysql.socket",
+		prefix + "name":     "grafana",
+		prefix + "user":     "user",
+		prefix + "password": "password",
+	}, prefix)
+}
+
+func TestGetEngineMySQLFromConfig(t *testing.T) {
+	t.Parallel()
+
+	t.Run("happy path - with key prefix", func(t *testing.T) {
+		t.Parallel()
+		engine, err := getEngineMySQL(newValidMySQLGetter(true))
+		require.NotNil(t, engine)
+		require.NoError(t, err)
+	})
+
+	t.Run("happy path - without key prefix", func(t *testing.T) {
+		t.Parallel()
+		engine, err := getEngineMySQL(newValidMySQLGetter(false))
+		require.NotNil(t, engine)
+		require.NoError(t, err)
+	})
+
+	t.Run("invalid string", func(t *testing.T) {
+		t.Parallel()
+
+		getter := newTestConfGetter(map[string]string{
+			"db_type":     dbTypeMySQL,
+			"db_host":     "/var/run/mysql.socket",
+			"db_name":     string(invalidUTF8ByteSequence),
+			"db_user":     "user",
+			"db_password": "password",
+		}, "db_")
+		engine, err := getEngineMySQL(getter)
+		require.Nil(t, engine)
+		require.Error(t, err)
+		require.ErrorIs(t, err, errInvalidUTF8Sequence)
+	})
+}
+
+func newValidPostgresGetter(withKeyPrefix bool) confGetter {
+	var prefix string
+	if withKeyPrefix {
+		prefix = "db_"
+	}
+	return newTestConfGetter(map[string]string{
+		prefix + "type":     dbTypePostgres,
+		prefix + "host":     "localhost",
+		prefix + "name":     "grafana",
+		prefix + "user":     "user",
+		prefix + "password": "password",
+	}, prefix)
+}
+
+func TestGetEnginePostgresFromConfig(t *testing.T) {
+	t.Parallel()
+
+	t.Run("happy path - with key prefix", func(t *testing.T) {
+		t.Parallel()
+		engine, err := getEnginePostgres(newValidPostgresGetter(true))
+		require.NotNil(t, engine)
+		require.NoError(t, err)
+	})
+
+	t.Run("happy path - without key prefix", func(t *testing.T) {
+		t.Parallel()
+		engine, err := getEnginePostgres(newValidPostgresGetter(false))
+		require.NotNil(t, engine)
+		require.NoError(t, err)
+	})
+
+	t.Run("invalid string", func(t *testing.T) {
+		t.Parallel()
+		getter := newTestConfGetter(map[string]string{
+			"db_type":     dbTypePostgres,
+			"db_host":     string(invalidUTF8ByteSequence),
+			"db_name":     "grafana",
+			"db_user":     "user",
+			"db_password": "password",
+		}, "db_")
+		engine, err := getEnginePostgres(getter)
+
+		require.Nil(t, engine)
+		require.Error(t, err)
+	})
+
+	t.Run("invalid hostport", func(t *testing.T) {
+		t.Parallel()
+		getter := newTestConfGetter(map[string]string{
+			"db_type":     dbTypePostgres,
+			"db_host":     "1:1:1",
+			"db_name":     "grafana",
+			"db_user":     "user",
+			"db_password": "password",
+		}, "db_")
+		engine, err := getEnginePostgres(getter)
+
+		require.Nil(t, engine)
+		require.Error(t, err)
+	})
+}
+
+func TestGetEngineMySQLTLS(t *testing.T) {
```

Comment:
Related to the above comment in the implementation, I think we need a whole `TestGetEnginePostgresTLS` given the config options are so different.